### PR TITLE
Testing rack swapping

### PR DESCRIPTION
**Pull request by carlyeks**
For CASSANDRA-10242

### CODE REVIEW

File: replication_test.py [hunk 1]
```diff
@@ -456,6 +456,37 @@ def test_rf_expand_yaml_file_snitch(self):
...
+
+        node.stop()
+
+        with open(os.path.join(node.get_conf_dir(), 'cassandra-rackdc.properties'), 'w') as topo_file:
+            topo_file.write("dc=dc1" + os.linesep)
+            topo_file.write("rack=rack0" + os.linesep)
+
+        mark = node.mark_log()
+        node.start()
+        node.watch_log_for(expected_error, from_mark=mark, timeout=10)
```
Comment (mambocab) [inline replication_test.py]: Can you add an assertion that the node isn't running at the end here? Maybe after a few seconds' sleep. @ptnapoleon do we have a standard way to check that startup failed?

Comment (ptnapoleon) [inline replication_test.py]: node.is_running() is a pretty good one.

**Comment by ptnapoleon** [comment]
Stefania just submitted a much larger PR for 10242, can we close this in favor of that? #688

**Comment by mambocab** [comment]
I have a few style nits, most noted inline, but I'd rather merge this and get the tests passing, then fix up the style stuff myself. Thanks for the tests stefania! Merging.

**Comment by mambocab** [comment]
Sorry, all, I merged this thinking I was looking at #688. Is it made redundant by that PR?

**Comment by ptnapoleon** [comment]
Moved to #695, sorry.

**Pull request merged**